liblsp/protocol_removed_structs.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+// XXX this file is purely kept in case I need some of those structs later. Remove file before merge.
+#include <liblsp/protocol.h>
+
+namespace lsp::protocol {
```

Comment:
I think we should go with placing `lsp` under `solidity` as we do with everything. I understand you say that lsp is more generic, but unless it is split out to be a third-party library, it is really just part of the solidity projecy.